### ISSUE TITLE
Does the first element of the list generated by the modular-scale-list function serve a legitimate purpose?

### ISSUE THREAD

**Issue by cimmanon** [modularscale/modularscale-sass#54]
I'm using modular-scale to generate the font-sizes used by my headlines (h1-h6).  When I specified `modular-scale-list($n, $n + $i)`, my expectation as a programmer is that I would have a list containing `$i + 1` items.  Instead, I receive a list of `$i + 2` items where the first item is the string `MS-LIST:`.

I understand that this string is desirable for debugging purposes with the modular-scale-list mixin, but wouldn't it make more sense to have that string be contained within the mixin instead of returned by the function?

**Comment by scottkellum** [comment]
What is your use case for needing a list other than debugging?

**Comment by cimmanon** [comment]
``` scss
@import "modular-scale";

$heading-font-family: $optima !default;
$heading-scale-start: -2 !default;
$heading-scale: major-third() !default;

$heading-size-list: modular-scale-list($heading-scale-start, $heading-scale-start + 5, 1em, $heading-scale);

@include headings {
    font-family: $heading-font-family;
    font-weight: 700;
}

@if true { // new scope to keep $i from being a global variable
    $i: 6;
    @each $s in $heading-size-list {
        @if $s != unquote("MS-LIST:") { // why is this here?
            h#{$i} {
                font-size: $s;
            }
            $i: $i - 1;
        }
    }
}
```

**Comment by scottkellum** [comment]
Because the ms() function is already so easy to manipulate I never considered using `modular-scale-list` programmatically, only for clear debug output. I see your point, I could be convinced to change this.

Here is how I might do it:

``` scss
@import "modular-scale";

$heading-font-family: $optima !default;
$heading-scale-start: -2 !default;
$heading-scale: major-third() !default;

@include headings {
    font-family: $heading-font-family;
    font-weight: 700;
}

$heading-scale-start: -2 !default;
$heading-scale: major-third() !default;

@for $i from 1 through 5 {
    h#{$i} {
        font-size: ms($heading-scale-start + abs($i - 5));
    }
}
```

**Comment by cimmanon** [comment]
Personally, I don't find the use of abs very intuitive.  If loops could decrement (see: https://github.com/nex3/sass/issues/691) or if there was a reverse function for lists, then it would be simpler.

``` scss
$heading-size-list: modular-scale-list($heading-scale-start + 5, $heading-scale-start, 1em, $heading-scale);

@for $i from 1 through 6 {
    h#{$i} {
        font-size: nth($heading-size-list, $i);
    }
}
```

**Comment by scottkellum** [comment]
@cimmanon yeah, plenty of ways to write it though, shouldn’t be too hard to write it with the loop you have.

Keeping this issue open, I’ll probably remove the `MS-LIST:` but am totally swamped with things at the moment.

**Comment by scottkellum** [comment]
[Should be fixed in 2.0](https://github.com/Team-Sass/modular-scale/blob/2.x/readme.md). Please reopen if it isn’t.
